architecture/protocol.go

```diff
@@ -147,6 +147,16 @@ func (command *Command) Parse(rawCommand string) (bool, error) {
 			return true, command.Err
 		}
 
+		// Check if parameter types are correct
+		for i, _ := range opts.Params {
+			if opts.ParamInteger[i] {
+				_, err := strconv.Atoi(parts[i+1])
```

Comment:
@marekyggdrasil sorry if I confused you with my comment, but its not so much a requirement for this pull req. I won't mind for now if we just have the validation done in a configurable way. As long as @stephenwithavs concerns are addressed in this pull req I don't mind getting this merged.